src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/WorkTypesControllerTest.java

```diff
@@ -0,0 +1,214 @@
+
+    @Mock
+    private AccessControlResponse mockedAccessControlResponse;
+
+    private WorkTypesController workTypesController;
+
+    @BeforeEach
+    void setUp() {
+        workTypesController = new WorkTypesController(
+            taskManagementService,
+            accessControlService
+        );
+    }
+
+    @Test
+    void should_return_a_single_element_list_of_work_type_when_user_has_a_single_work_type() {
+
+        UserInfo userInfo = new UserInfo("", "",
+                                         new ArrayList<>(Arrays.asList("Role1","Role2")),
+                                         "",
+                                         "",
+                                         "");
+
+        RoleAssignment roleAssignment = new RoleAssignment(ActorIdType.IDAM,
+                                                           "1258555",
+                                                           RoleType.CASE,
+                                                           "Judge",
+                                                           Classification.PUBLIC,
+                                                           GrantType.BASIC,
+                                                           RoleCategory.JUDICIAL,
+                                                           false,
+                                                           Map.of("workTypes","hearing_work"));
+        List<RoleAssignment> roleAssignmentList = Arrays.asList(roleAssignment);
+        AccessControlResponse accessControlResponse = new AccessControlResponse(userInfo,roleAssignmentList);
+
+        when(accessControlService.getRoles(IDAM_AUTH_TOKEN))
+            .thenReturn(accessControlResponse);
+
+        when(accessControlService.getRolesByActorId(roleAssignment.getActorId(),IDAM_AUTH_TOKEN))
+            .thenReturn(accessControlResponse);
+
+        WorkType workType = new WorkType("hearing_work","Hearing Work");
+
+        when(taskManagementService.getWorkType(
+            workType.getId()
+        ))
+            .thenReturn(Optional.of(workType));
+
+        ResponseEntity<List<WorkType>> response = workTypesController.getWorkTypes(IDAM_AUTH_TOKEN);
+
+        assertNotNull(response);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertThat(response.getBody(), instanceOf(ArrayList.class));
+        assertNotNull(response.getBody());
+
+        List<WorkType> workTypeList = Arrays.asList(workType);
+        assertEquals(workTypeList, response.getBody());
+        verify(taskManagementService, times(1))
+            .getWorkType(workType.getId());
+    }
+
+    @Test
+    void should_return_a_list_of_work_types_when_user_has_many_work_types() {
+
+        UserInfo userInfo = new UserInfo("", "",
+                                         new ArrayList<>(Arrays.asList("Role1","Role2")),
+                                         "",
+                                         "",
+                                         "");
```

Comment:
you should be able to use the UserInfo.builder which makes it cleaner

Reply:
used userBuilder.